src/commissioner/commissioner_api.h

```diff
@@ -0,0 +1,224 @@
+/*
+ *    Copyright (c) 2019, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   The file defines commissioner c api
+ */
+
+#ifndef OTBR_COMMISSIONER_C_API_H_
+#define OTBR_COMMISSIONER_C_API_H_
+
+#include <stdint.h>
+#include <string.h>
+
+#include <arpa/inet.h>
+#include <sys/socket.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum otbr_commissioner_result_t
+{
+    OTBR_COMMISSIONER_SUCCESS        = 0,
+    OTBR_COMMISSIONER_INTERNAL_ERROR = -1,
+    OTBR_COMMISSIONER_INVALID_ARGS   = -0xffff
+};
+
+typedef void *otbr_commissioner_handle_t;
+typedef void *steering_data_handle_t;
+
+/**
+ * This function creates steering data to filter joiners.
+ *
+ * @param[out]   aSteeringData          Buffer to store steering data handle.
+ * @param[in]    aSteeringDataLength    Length of steering data.
+ * @param[in]    aAllowAll              Whether to allow devices with any eui64 to join.
+ * @param[in]    aJoinerEui64           Target joiner eui64, only used when aAllowAll==false.
+ *
+ * @retval OTBR_COMMISSIONER_INVALID_ARGS if nullptr provided, OTBR_COMMISSIONER_SUCCESS otherwise.
```

Comment:
Should have a separate line for each return value. Also should document which parameters are checked.
```
* @retval OTBR_COMMISSIONER_SUCCESS if steering data was successfully created.
* @retval OTBR_COMMISSIONER_INVALID_ARGS if nullptr was provided for @p aSteeringData.
```
```